src/queue.ts

```diff
@@ -21,18 +34,29 @@ const startQueue = <T extends EventBase>(
 ) => {
   const queue: number[] = [];
   const dedupeSet = new Set<number>();
+  const processedSet = new Set<number>();
   const promiseMap = new Map<number, PromiseResolver>();
   let state: 'READY' | 'PROCESSING' = 'READY';
 
   const processEvent = (event: EventBase) => {
-    reduxStore.dispatch(event);
     if (event.id === undefined) {
       throw new Error(`Malformed event is missing id: ${event}`);
     }
+    if (processedSet.has(event.id)) {
+      return;
+    }
+    reduxStore.dispatch(event);
     const resolver = promiseMap.get(event.id);
+    processedSet.add(event.id);
     if (resolver) {
       resolver(reduxStore.getState());
-      promiseMap.delete(event.id);
+    }
+
+    // Cleanup
+    promiseMap.delete(event.id);
+    if (event.id % 100_000 === 0) {
```

Comment:
I'm guessing these values are mostly arbitrary, would it be worth setting them to variables just to make it a bit clearer why we're doing it?

Reply:
Yeah I want to include them in an optional config object provided when initialising things, but didn't get to it yet. Atm these are arbitrary numbers to ensure the sets don't grow infinitely